cardano_node_tests/tests/test_pools.py

```diff
@@ -337,6 +337,117 @@ def _deregister_stake_pool_w_build(
+        src_address=pool_owners[0].payment.address,
+        tx_name=tx_name,
+        tx_files=clusterlib.TxFiles(
+            certificate_files=[pool_reg_cert_file],
+            signing_key_files=signing_key_files,
+        ),
+        deposit=deposit,
+        # witness_count_add instead of override (estimate needs this hint)
+        witness_count_add=len(signing_key_files),
+        destination_dir=destination_dir,
+    )
+
+    tx_signed = cluster_obj.g_transaction.sign_tx(
+        tx_body_file=tx_raw_output.out_file,
+        signing_key_files=signing_key_files[:1],
+        tx_name=f"{tx_name}_sign0",
+    )
+    tx_signed_inc = cluster_obj.g_transaction.sign_tx(
+        tx_file=tx_signed,
+        signing_key_files=signing_key_files[1:],
+        tx_name=f"{tx_name}_sign1",
+    )
+
+    cluster_obj.g_transaction.submit_tx(tx_file=tx_signed_inc, txins=tx_raw_output.txins)
+
+    dbsync_utils.check_tx(cluster_obj=cluster_obj, tx_raw_output=tx_raw_output)
+
+    return pool_reg_cert_file, tx_raw_output
+
+
+def _create_stake_pool_w_build_est(
+    cluster_obj: clusterlib.ClusterLib,
+    pool_data: clusterlib.PoolData,
+    pool_owners: list[clusterlib.PoolUser],
+    tx_name: str,
+    reward_account_key_pair: clusterlib.KeyPair | clusterlib.AddressRecord | None = None,
+    destination_dir: clusterlib.FileType = ".",
+) -> clusterlib.PoolCreationOutput:
+    """Create and register a stake pool using a `transaction build-estimate` command."""
+    node_kes = cluster_obj.g_node.gen_kes_key_pair(
+        node_name=pool_data.pool_name,
+        destination_dir=destination_dir,
+    )
+    node_vrf = cluster_obj.g_node.gen_vrf_key_pair(
+        node_name=pool_data.pool_name,
+        destination_dir=destination_dir,
+    )
+    node_cold = cluster_obj.g_node.gen_cold_key_pair_and_counter(
+        node_name=pool_data.pool_name,
+        destination_dir=destination_dir,
+    )
+
+    # Register pool using build-estimate
+    pool_reg_cert_file, tx_raw_output = _register_stake_pool_w_build_est(
+        cluster_obj=cluster_obj,
+        pool_data=pool_data,
+        pool_owners=pool_owners,
+        vrf_vkey_file=node_vrf.vkey_file,
+        cold_key_pair=node_cold,
+        tx_name=tx_name,
+        reward_account_vkey_file=(
+            reward_account_key_pair.vkey_file if reward_account_key_pair else None
+        ),
+        destination_dir=destination_dir,
+    )
+
+    dbsync_utils.check_tx(cluster_obj=cluster_obj, tx_raw_output=tx_raw_output)
+
+    return clusterlib.PoolCreationOutput(
+        stake_pool_id=cluster_obj.g_stake_pool.get_stake_pool_id(node_cold.vkey_file),
+        vrf_key_pair=node_vrf,
+        cold_key_pair=node_cold,
+        pool_reg_cert_file=pool_reg_cert_file,
+        pool_data=pool_data,
+        pool_owners=pool_owners,
+        reward_account_key_pair=reward_account_key_pair or pool_owners[0].stake,
+        tx_raw_output=tx_raw_output,
+        kes_key_pair=node_kes,
+    )
+
+
```

Comment:
Please adapt and rename existing `_register_stake_pool_w_build` and `_create_stake_pool_w_build` to work with both `build` and `build-estimate`. This will reduce duplication.